contracts/contracts/MessageProcessor.sol

```diff
@@ -82,51 +84,36 @@ contract MessageProcessor is Ownable, SnarkCommon, Hasher, CommonUtilities, IMes
       revert NoMoreMessages();
     }
 
-    // The state AccQueue must be merged
-    if (!poll.stateMerged()) {
-      revert StateNotMerged();
-    }
-
     // Retrieve stored vals
-    (, uint8 messageTreeSubDepth, uint8 messageTreeDepth, uint8 voteOptionTreeDepth) = poll.treeDepths();
+    (, uint8 voteOptionTreeDepth) = poll.treeDepths();
     // calculate the message batch size from the message tree subdepth
-    uint256 messageBatchSize = TREE_ARITY ** messageTreeSubDepth;
-
-    (, AccQueue messageAq) = poll.extContracts();
-
-    // Require that the message queue has been merged
-    uint256 messageRoot = messageAq.getMainRoot(messageTreeDepth);
-    if (messageRoot == 0) {
-      revert MessageAqNotMerged();
-    }
+    uint256 messageBatchSize = poll.batchSizes();
```

Comment:
let's add `messageBatchSize` as a constructor parameter (to be immutable) to the MessageProcessor contract so we don't need to fetch it every time